core/src/main/java/org/elasticsearch/rest/BaseRestHandler.java

```diff
@@ -59,16 +64,40 @@ public final void handleRequest(RestRequest request, RestChannel channel, NodeCl
         final RestChannelConsumer action = prepareRequest(request, client);
 
         // validate unconsumed params, but we must exclude params used to format the response
-        final List<String> unconsumedParams =
-            request.unconsumedParams().stream().filter(p -> !responseParams().contains(p)).collect(Collectors.toList());
+        // use a sorted set so the unconsumed parameters appear in a reliable sorted order
+        final SortedSet<String> unconsumedParams =
+            request.unconsumedParams().stream().filter(p -> !responseParams().contains(p)).collect(Collectors.toCollection(TreeSet::new));
 
         // validate the non-response params
         if (!unconsumedParams.isEmpty()) {
-            final String message = String.format(
-                Locale.ROOT,
-                "request [%s] contains unrecognized parameters: %s",
-                request.path(),
-                unconsumedParams.toString());
+            String message = "request [" + request.path() + "] contains unrecognized parameters: ";
+            boolean first = true;
+            for (final String unconsumedParam : unconsumedParams) {
+                final LevensteinDistance ld = new LevensteinDistance();
+                final List<Tuple<Float, String>> scoredParams = new ArrayList<>();
+                for (String consumedParam : request.consumedParams()) {
+                    final float distance = ld.getDistance(unconsumedParam, consumedParam);
+                    if (distance > 0.5f) {
```

Comment:
I don't think so. The URL parameters tend to be shorter than settings keys. If we use the same distance, we will miss candidates for typos like "flied" for "field" because even though there is only one transposition, the distance in this case is 0.6.